--- a/qubes-gui-agent-windows-4.0.0/qvideo/gdi/driver.h	2018-07-11 16:29:59.000000000 +0000
+++ b/qubes-gui-agent-windows-4.0.0/qvideo/gdi/driver.h	2019-06-04 13:41:19.325480147 +0000
@@ -30,6 +30,13 @@
 #include <winddi.h>
 #include <devioctl.h>
 #include <ntddvdeo.h>
+#ifdef __MINGW__
+#include <driverspecs.h>
+
+#define __in_opt SAL__in_opt
+#define __out_bcount_opt SAL__out_bcount_opt
+#define __inout_bcount SAL__inout_bcount
+#endif
 
 #pragma warning(pop)
 // C4200: nonstandard extension used :
