--- a/qubes-core-agent-windows-4.0.0/src/qrexec-services/file-receiver/wdk.h    2018-07-11 14:32:41.000000000 +0000
+++ b/qubes-core-agent-windows-4.0.0/src/qrexec-services/file-receiver/wdk.h    2019-06-03 15:16:22.516472873 +0000
@@ -23,6 +23,10 @@

 typedef LONG NTSTATUS, *PNTSTATUS;

+#ifdef __MINGW__
+#define __field_bcount_part(size,init) SAL__notnull SAL__byte_writableTo(size) SAL__byte_readableTo(init)
+#endif
+
 typedef struct _UNICODE_STRING
 {
     USHORT Length;
