--- a/qubes-windows-utils-4.0.0/src/pipe-server.c       2018-07-11 14:28:13.000000000 +0000
+++ b/qubes-windows-utils-4.0.0/src/pipe-server.c       2019-06-03 14:38:06.795181175 +0000
@@ -74,7 +74,7 @@
 };

 // Initialize data for a newly connected client.
-DWORD QpsConnectClient(
+static DWORD QpsConnectClient(
     IN  PIPE_SERVER Server,
     IN  LONGLONG ClientId,
     IN  HANDLE WritePipe,
@@ -83,7 +83,7 @@

 // Disconnect can be called from inside worker threads, we don't want to
 // wait for them in that case.
-void QpsDisconnectClientInternal(
+static void QpsDisconnectClientInternal(
     IN  PIPE_SERVER Server,
     IN  LONGLONG ClientId,
     IN  BOOL WriterExiting,
--- a/qubes-windows-utils-4.0.0/src/service.c   2018-07-11 14:28:13.000000000 +0000
+++ b/qubes-windows-utils-4.0.0/src/service.c   2019-06-03 14:38:06.795181175 +0000
@@ -41,8 +41,8 @@
 // since ServiceMain doesn't take any context parameter.
 PSERVICE_CONTEXT g_Service = NULL;

-void SvcStop(void);
-void WINAPI SvcMain(DWORD argc, WCHAR *argv[]);
+static void SvcStop(void);
+static void WINAPI SvcMain(DWORD argc, WCHAR *argv[]);

 static void SvcSetState(
     IN DWORD state,
--- a/qubes-windows-utils-4.0.0/include/list.h  2018-07-11 14:28:13.000000000 +0000
+++ b/qubes-windows-utils-4.0.0/include/list.h  2019-06-03 14:38:06.794181173 +0000
@@ -21,6 +21,9 @@

 #pragma once
 #include <winnt.h>
+#ifdef __MINGW__
+#include <driverspecs.h>
+#endif

 // List macros from Windows kernel headers

@@ -37,7 +40,7 @@

 _Must_inspect_result_
 BOOLEAN
-CFORCEINLINE
+FORCEINLINE
 IsListEmpty(
     _In_ const LIST_ENTRY * ListHead
     )
