--- a/qubes-gui-agent-windows-4.0.0/qvideo/miniport/qvmini.h	2019-06-04 14:18:02.126557632 +0000
+++ b/qubes-gui-agent-windows-4.0.0/qvideo/miniport/qvmini.h	2019-06-04 14:17:58.091548326 +0000
@@ -38,7 +38,8 @@
 #include "list.h"
 #include "memory.h"
 
-#define QFN "[QVMINI] " __FUNCTION__ ": "
+#define conc(str1,str2) #str1 " " #str2
+#define QFN "[QVMINI] " conc(__FUNCTION__,": ")
 
 // device extension, per-adapter data
 typedef struct _QVMINI_DX
