--- a/qubes-gui-agent-windows-4.0.0/qvideo/gdi/support.h        2018-07-11 16:29:59.000000000 +0000
+++ b/qubes-gui-agent-windows-4.0.0/qvideo/gdi/support.h        2019-06-03 16:38:58.548896693 +0000
@@ -41,14 +41,16 @@
 // http://msdn.microsoft.com/en-us/library/windows/hardware/ff551519(v=vs.85).aspx
 #define _DEBUGF(format, ...) DbgPrintEx(DPFLTR_IHVVIDEO_ID, DPFLTR_TRACE_LEVEL, format, ##__VA_ARGS__)

+#define conc(str1,str2) #str1 " " #str2
+
 #define TRACEF(format, ...) DbgPrintEx(DPFLTR_IHVVIDEO_ID, DPFLTR_INFO_LEVEL, \
-    "[" DRIVER_NAME "] " __FUNCTION__ ": " format "\n", ##__VA_ARGS__)
+    "[" DRIVER_NAME "] " conc(__FUNCTION__,": ") format "\n", ##__VA_ARGS__)
 #define DEBUGF(format, ...) DbgPrintEx(DPFLTR_IHVVIDEO_ID, DPFLTR_TRACE_LEVEL, \
-    "[" DRIVER_NAME "] " __FUNCTION__ ": " format "\n", ##__VA_ARGS__)
+    "[" DRIVER_NAME "] " conc(__FUNCTION__,": ") format "\n", ##__VA_ARGS__)
 #define WARNINGF(format, ...) DbgPrintEx(DPFLTR_IHVVIDEO_ID, DPFLTR_WARNING_LEVEL, \
-    "[" DRIVER_NAME "] " __FUNCTION__ ": " format "\n", ##__VA_ARGS__)
+    "[" DRIVER_NAME "] " conc(__FUNCTION__,": ") format "\n", ##__VA_ARGS__)
 #define ERRORF(format, ...) DbgPrintEx(DPFLTR_IHVVIDEO_ID, DPFLTR_ERROR_LEVEL, \
-    "[" DRIVER_NAME "] " __FUNCTION__ ": " format "\n", ##__VA_ARGS__)
+    "[" DRIVER_NAME "] " conc(__FUNCTION__,": ") format "\n", ##__VA_ARGS__)

 #define FUNCTION_ENTER()   TRACEF("==>")
 #define FUNCTION_EXIT()    TRACEF("<==")
