--- a/qubes-gui-agent-windows-4.0.0/include/version_common.rc   2018-07-11 16:29:59.000000000 +0000
+++ b/qubes-gui-agent-windows-4.0.0/include/version_common.rc   2019-06-03 16:18:50.820112862 +0000
@@ -3,7 +3,9 @@

 // version.h is generated during build by set_version.ps1
 // it defines version macros
+#ifndef __MINGW__
 #include "..\version.h"
+#endif

 // QTW_FILETYPE must be defined (VFT_*)

