--- a/qubes-gui-agent-windows-4.0.0/qvideo/gdi/enable.c	2018-07-11 16:29:59.000000000 +0000
+++ b/qubes-gui-agent-windows-4.0.0/qvideo/gdi/enable.c	2019-06-04 14:13:16.561899403 +0000
@@ -26,12 +26,14 @@
 
 #define STATUS_SUCCESS 0
 
+#ifndef __MINGW__
 // This is only to make linker happy. Enabling security checks
 // causes linking a special library that expects that DriverEntry is present.
 void DriverEntry()
 {
     // THIS SHOULD NOT BE CALLED, EVER
 }
+#endif
 
 // The driver function table with all function index/address pairs
 static DRVFN g_DrvFunctions[] =
